compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/UKernelOps.cpp

```diff
@@ -184,22 +185,24 @@ UKernelGenericOp::lowerToFunctionCall(RewriterBase &rewriter) {
 void UKernelGenericOp::getEffects(
     SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
         &effects) {
-  SmallVector<Value> readOnlyOperands = getDpsInputs();
-  readOnlyOperands.append(getOtherOperands().begin(), getOtherOperands().end());
-  for (Value value : readOnlyOperands) {
-    if (!llvm::isa<MemRefType>(value.getType())) {
+  SmallVector<OpOperand *> readOnlyOperands = getDpsInputOperands();
+  for (OpOperand &operand : getOtherOperandsMutable()) {
+    readOnlyOperands.push_back(&operand);
+  }
```

Comment:
Since you are updating this anyway, we can do:
```suggestion
  llvm::append_range(readOnlyOperands, llvm::make_pointer_range(getOtherOperandsMutable()));
```